### ISSUE TITLE
GCStress: Crash in 10w5d_cs_d.exe Linux

### ISSUE THREAD

**Issue by swaroop-sridhar** [dotnet/coreclr#2757]
Repro steps are:
* cp \\scratch2\scratch\swaroops\coredistools\libcoredistools.so somewhere 
* export LD_LIBRARY_PATH=somewhere
* export COMPlus_GCStress=0xC
* Run the test below

```
corerun JIT/Methodical/fp/exgen/10w5d_cs_d/10w5d_cs_d.exe

Assert failure(PID 18927 [0x000049ef], Thread: 18927 [0x49ef]): m_alignpad == 0
    File: /home/swaroops/git/a/coreclr/src/vm/syncblk.cpp Line: 3660
    Image: /home/swaroops/git/a/coreclr/bin/Product/Linux.x64.Debug/corerun

**** MessageBox invoked, title 'corerun - Assert Failure (PID 18927, Thread 18927/0x49ef)' ****
  m_alignpad == 0

/home/swaroops/git/a/coreclr/src/vm/syncblk.cpp, Line: 3660

Abort - Kill program
Retry - Debug
Ignore - Keep running

Image:
/home/swaroops/git/a/coreclr/bin/Product/Linux.x64.Debug/corerun

********
```

**Comment by tzwlai** [comment]
CC @sergiy-k @adityamandaleeka

**Comment by swaroop-sridhar** [comment]
Similar failure in `JIT/Methodical/fp/exgen/10w5d_cs_r/10w5d_cs_r.exe` likely to be the same cause.
```
corerun JIT/Methodical/fp/exgen/10w5d_cs_r/10w5d_cs_r.exe

Assert failure(PID 17788 [0x0000457c], Thread: 17788 [0x457c]): m_alignpad == 0
    File: /home/swaroops/git/a/coreclr/src/vm/syncblk.cpp Line: 3660
    Image: /home/swaroops/git/a/coreclr/bin/Product/Linux.x64.Debug/corerun

**** MessageBox invoked, title 'corerun - Assert Failure (PID 17788, Thread 17788/0x457c)' ****
  m_alignpad == 0

/home/swaroops/git/a/coreclr/src/vm/syncblk.cpp, Line: 3660
Abort - Kill program
Retry - Debug
Ignore - Keep running

Image:
/home/swaroops/git/a/coreclr/bin/Product/Linux.x64.Debug/corerun

********
```

**Comment by swaroop-sridhar** [comment]
Similarly in binarytrees.exe

```
corerun JIT/Performance/CodeQuality/BenchmarksGame/binarytrees/binarytrees/binarytrees.exe

Assert failure(PID 23591 [0x00005c27], Thread: 23591 [0x5c27]): m_alignpad == 0
    File: /home/swaroops/git/a/coreclr/src/vm/syncblk.cpp Line: 3660
    Image: /home/swaroops/git/a/coreclr/bin/Product/Linux.x64.Debug/corerun

**** MessageBox invoked, title 'corerun - Assert Failure (PID 23591, Thread 23591/0x5c27)' ****
  m_alignpad == 0

/home/swaroops/git/a/coreclr/src/vm/syncblk.cpp, Line: 3660

Abort - Kill program
Retry - Debug
Ignore - Keep running

Image:
/home/swaroops/git/a/coreclr/bin/Product/Linux.x64.Debug/corerun

********
```

**Comment by sivarv** [comment]
This is likely be a gc-hole jitted code.  After ruling that out will assign to VM team if required.

**Comment by sivarv** [comment]
Investigation reveals two more gc-holes while passing and returning structs (containing gc-refs) in two registers.

Struct return from a call:
Codegen needs to inform emitter that one or both of the return registers contains gc-ref/by-ref and emitter in turn needs to emit gc-info accordingly.

Struct passing in two registers:
genCallInstruction() needs to specify the right emit type while passing individual fields struct passed in two registers.

**Comment by sivarv** [comment]
Assigning t @LLITCHEV  as he is working on a fix for this.

**Comment by LLITCHEV** [comment]
Fixed by PR #3351 .
